sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1985,6 +1985,75 @@ static void ReverseRangeDimensionsForKernel(NDRDescT &NDR) {
   }
 }
 
+// Sets arguments for a given kernel and device based on the argument type.
+// Refactored from SetKernelParamsAndLaunch to allow it to be used in the graphs
+// extension.
+void SetArgBasedOnType(
+    const detail::plugin &Plugin, RT::PiKernel Kernel,
+    const std::shared_ptr<device_image_impl> &DeviceImageImpl,
+    const std::function<void *(Requirement *Req)> &getMemAllocationFunc,
+    const sycl::context &Context, bool IsHost, detail::ArgDesc &Arg,
+    size_t NextTrueIndex) {
+  switch (Arg.MType) {
+  case kernel_param_kind_t::kind_stream:
+    break;
+  case kernel_param_kind_t::kind_accessor: {
+    Requirement *Req = (Requirement *)(Arg.MPtr);
+    if (getMemAllocationFunc == nullptr)
+      throw sycl::exception(make_error_code(errc::kernel_argument),
+                            "placeholder accessor must be bound by calling "
+                            "handler::require() before it can be used.");
+
+    RT::PiMem MemArg = (RT::PiMem)getMemAllocationFunc(Req);
+    if (Plugin.getBackend() == backend::opencl) {
+      Plugin.call<PiApiKind::piKernelSetArg>(Kernel, NextTrueIndex,
+                                             sizeof(RT::PiMem), &MemArg);
+    } else {
+      Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, NextTrueIndex,
+                                                      &MemArg);
+    }
+    break;
+  }
+  case kernel_param_kind_t::kind_std_layout: {
```

Comment:
Hmm, I'm not seeing this error when I run a debug build. Is this in Release mode for either the library or example?

Reply:
Thanks I can repro now using `-O2` to build the example, will look into it on Monday!

Reply:
This should be fixed now! I hadn't implemented the arg filtering and because the float arg is `const` in the saxpy test it must have been eliminated with optimizations on.